Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.Exam Scores

In [23]:
# Creating a Sample Dataset
studying_time = [2,3,1,4,5,6,5,9,3,3]
exam_scores = [70,75,60,73,78,79,80,90,65,63]

In [16]:
import pandas as pd
df = pd.DataFrame({'Studying_time' : studying_time, 'Exam Scores':exam_scores})

In [20]:
p = df.corr(method='pearson')['Studying_time']['Exam Scores']

In [22]:
print(f'Pearson Correlation Coefficient for between Studying Time and Final Exam Scores for the given data is: {round(p,3)}')

Pearson Correlation Coefficient for between Studying Time and Final Exam Scores for the given data is: 0.908


##### The value of 0.908 indicates that there's a strong, positive correlation between studying time and final exam scores.

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [1]:
# Creating a Sample Dataset
import pandas as pd
import numpy as np

n = 50 # Sample size
np.random.seed(120)

df = pd.DataFrame({'Amount of Sleep': np.random.randint(4, 11, n), 'Job Satisfaction': np.random.randint(1, 11, n)})

df.head()

Amount of Sleep  Job Satisfaction
0                4                 5
1               10                 3
2               10                 2
3                6                 7
4                4                10

In [2]:
df.corr(method='spearman')

Amount of Sleep  Job Satisfaction
Amount of Sleep          1.000000         -0.199812
Job Satisfaction        -0.199812          1.000000

##### Spearman's rank correlation coefficient of -0.2 indicates a weak, negative monotonic relationship between the amount of sleep individuals get each night and their overall job satisfaction level.

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

In [6]:
import pandas as pd
import numpy as np

n = 50 # Sample size
np.random.seed(123)

df = pd.DataFrame({'No of Hours': np.random.randint(6, 15, n), 'BMI': np.random.randint(18, 45, n)})

df.head()

No of Hours  BMI
0            8   29
1            8   40
2           12   25
3            7   19
4            9   29

In [7]:
df.corr(method='pearson')

No of Hours       BMI
No of Hours     1.000000  0.159045
BMI             0.159045  1.000000

In [8]:
df.corr(method='spearman')

No of Hours       BMI
No of Hours     1.000000  0.152531
BMI             0.152531  1.000000

#### The Pearson correlation coefficient and the Spearman's rank correlation coefficient both indicate a weak positive correlation between the number of hours of exercise per week and body mass index (BMI) in the sample of adults.

#### The Pearson correlation coefficient is 0.159045, which is positive and indicates a weak linear relationship between the two variables. The coefficient is close to zero, which suggests that the correlation is not very strong.

#### The Spearman's rank correlation coefficient is 0.152531, which is also positive and indicates a weak monotonic relationship between the two variables. This coefficient is also close to zero, which suggests that the correlation is weak.

Overall, these results suggest that there is a slight tendency for people who exercise more to have slightly higher BMI scores, but the relationship is not very strong.

Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [11]:
import numpy as np

np.random.seed(123)

hours_tv = np.random.normal(4, 2, 50)
hours_physical = np.random.normal(5, 2, 50)

df = pd.DataFrame({'Time Spent Watching TV': hours_tv, 'Physical Activity': hours_physical })

df.head()

Time Spent Watching TV  Physical Activity
0                1.828739           2.411829
1                5.994691           2.922424
2                4.565957           8.487424
3                0.987411           3.403875
4                2.842799           5.059366

In [12]:
df.corr('pearson')

Time Spent Watching TV  Physical Activity
Time Spent Watching TV                1.000000          -0.085241
Physical Activity                    -0.085241           1.000000

#### The Pearson correlation coefficient between the number of hours individuals spend watching television per day and their level of physical activity is -0.085241. This value indicates a weak negative correlation between the two variables.

#### A negative correlation means that as the number of hours of TV watched per day increases, the level of physical activity tends to decrease slightly. However, the correlation is weak, which means that the relationship between the two variables is not very strong.

Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:

Age(Years)     | Soft drink Preference

25             |         Coke

42             |         Pepsi

37             |      Mountain dew

19             |         Coke

31             |         Pepsi

28             |         Coke

#### Firstly Encoding the data of Soft Drink for above sample.
Since the sample is small, we use dictionary for encoding.

In [16]:
d = {
    'Coke': 1,
    'Pepsi': 2,
    'Mountain Dew': 3
}

drink = ['Coke', 'Pepsi', 'Mountain Dew', 'Coke', 'Pepsi','Coke']

drink_encoded = [d[i] for i in drink]

In [17]:
age = [25, 42, 37, 19, 31,28]

In [18]:
df = pd.DataFrame({'Age':age,'Soft Drink Preferred':drink_encoded})

In [19]:
df

Age  Soft Drink Preferred
0   25                     1
1   42                     2
2   37                     3
3   19                     1
4   31                     2
5   28                     1

In [20]:
df.corr(method='spearman')

Age  Soft Drink Preferred
Age                   1.000000              0.833238
Soft Drink Preferred  0.833238              1.000000

#### A Spearman's rank correlation coefficient value of 0.833238 indicates a strong positive monotonic relationship between two variables.

A value of 0.833238 indicates that as the values of one variable increase, the values of the other variable tend to increase as well, and vice versa. Moreover, the strength of this relationship is strong, as the coefficient value is close to 1. This suggests that the two variables are likely to be associated with each other, and that the relationship is not likely to be due to chance.

In [22]:
df.corr(method='pearson')

Age  Soft Drink Preferred
Age                   1.000000              0.758704
Soft Drink Preferred  0.758704              1.000000

#### A Pearson correlation coefficient value of 0.758704 indicates a strong positive linear relationship between two variables.

A value of 0.758704 indicates that there is a strong positive linear relationship between the two variables. As one variable increases, the other variable tends to increase as well, and vice versa. Moreover, the strength of this relationship is strong, as the coefficient value is close to 1. This suggests that the two variables are highly associated with each other, and that the relationship is not likely to be due to chance.

Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [23]:
np.random.seed(123)
df = pd.DataFrame({'Sales Calls / Day': np.random.randint(10,50,30),'Sales Made / Week': np.random.randint(2,15,30)})

In [24]:
df.head()

Sales Calls / Day  Sales Made / Week
0                 12                  7
1                 38                  8
2                 44                 13
3                 48                  4
4                 27                  3

In [25]:
df.corr('pearson')

Sales Calls / Day  Sales Made / Week
Sales Calls / Day           1.000000          -0.111541
Sales Made / Week          -0.111541           1.000000

#### A Pearson correlation coefficient value of -0.111541 indicates a weak negative linear relationship between two variables.